crates/holochain_p2p/tests/tests/node_messaging.rs

```diff
@@ -422,6 +532,47 @@ async fn test_get() {
     .unwrap();
 }
 
+#[tokio::test(flavor = "multi_thread")]
+async fn test_get_with_unresponsive_agents() {
+    let dna_hash = DnaHash::from_raw_36(vec![0; 36]);
+    let space = dna_hash.to_k2_space();
+    let handler = Arc::new(Handler::default());
+    let unresponsive_handler = Arc::new(UnresponsiveHandler);
+
+    let (_agent1, hc1, _) = spawn_test(dna_hash.clone(), handler.clone()).await;
+    let (_agent2, hc2, _) = spawn_test(dna_hash.clone(), unresponsive_handler.clone()).await;
+    let (_agent3, hc3, _) = spawn_test(dna_hash.clone(), unresponsive_handler).await;
+    let (_agent4, hc4, _) = spawn_test(dna_hash.clone(), handler).await;
+
+    hc1.test_set_full_arcs(space.clone()).await;
+    hc2.test_set_full_arcs(space.clone()).await;
+    hc3.test_set_full_arcs(space.clone()).await;
+    hc4.test_set_full_arcs(space.clone()).await;
+
+    tokio::time::timeout(std::time::Duration::from_secs(20), async {
+        loop {
+            tokio::time::sleep(std::time::Duration::from_millis(1)).await;
```

Comment:
```suggestion
            tokio::time::sleep(std::time::Duration::from_millis(10)).await;
```

Just a suggestion, I generally assume that really short sleeps take compute away from other work, depending on how much is running in the background. I guess here it's not much. But when you have tests running in parallel, keeping these loops reasonably zen is helpful I think.

Reply:
This is also just the value of the existing tests. I was assuming/hoping that the call would actually succeed on the first attempt when it runs successfully, so I didn't think the delay was an issue. I can add a constant for this, though, and change it for all the tests. Or maybe I can just try without the loop and see if it works.

Reply:
Done: 10e3d0a7981eddb216384f96d009d297636e3fa1